pythonPackages.python-jwt: init at 1.3.1

thornycrackers opened the pull request:
###### Description of changes

For packaging request https://github.com/NixOS/nixpkgs/issues/215312

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

NickCao commented on pkgs/development/python-modules/python-jwt/default.nix:
```diff
@@ -0,0 +1,36 @@
+  propagatedBuildInputs = [
+    cryptography
+    freezegun
+    pytest
```
`cryptography` should be the only propagatedBuildInputs: https://github.com/GehirnInc/python-jwt/blob/068db420c9ae957925daf0f5a2baa9319ac20c82/setup.py#L27

The rest might find their place in nativeCheckInputs (also there is pytestCheckHook)

thornycrackers replied:
Good call. I tried using pytestCheckHook but it threw usage errors. Instead of diving into it I just added them to nativeCheckInputs

NickCao commented:
Suggesting some changes:
```diff
diff --git a/pkgs/development/python-modules/python-jwt/default.nix b/pkgs/development/python-modules/python-jwt/default.nix
index bc41315ba05..b6c6804f43b 100644
--- a/pkgs/development/python-modules/python-jwt/default.nix
+++ b/pkgs/development/python-modules/python-jwt/default.nix
@@ -4,7 +4,8 @@
 , pythonOlder
 , cryptography
 , freezegun
-, pytest
+, pytestCheckHook
+, pytest-cov
 }:
 
 buildPythonPackage rec {
@@ -20,14 +21,27 @@ buildPythonPackage rec {
     sha256 = "sha256-N1J8yBVX/O+92cRp+q2gA2cFsd+C7JjUR9jo0VGoINg=";
   };
 
-  nativeCheckInputs = [ pytest freezegun ];
-  propagatedBuildInputs = [ cryptography ];
+  postPatch = ''
+    # pytest-flake8 is incompatible flake8 6.0.0 and currently unmaintained
+    substituteInPlace setup.cfg --replace "--flake8" ""
+  '';
+
+  propagatedBuildInputs = [
+    cryptography
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    freezegun
+    pytest-cov
+  ];
 
   pythonImportsCheck = [ "jwt" ];
 
   meta = with lib; {
     description = "JSON Web Token library for Python 3";
     homepage = "https://github.com/GehirnInc/python-jwt";
-    license = "Apache-2.0";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ];
   };
 }
```

NickCao commented:
`pytest-flake8` is unmaintained and currently marked broken, just patch that use out. Keeping the dependencies one per line would make future changes produce cleaner diffs. `meta.maintainers` must be set according to `CONTRIBUTING.md`, and preferably you may add yourself as a maintainer.

thornycrackers commented:
Well that makes sense why it was complaining about flake8. Thanks for your patience and explaining everything. Didn't know about `substituteInPlace`, that's a cool trick. I've updated all the suggestions and added myself as the maintainer.

NickCao commented:
And you should keep the changes in two commits:
1. `maintainers: add thornycrakers`
2. `pythonPackages.python-jwt: init at 1.3.1`

thornycrackers committed:
python3Packages.python-jwt: init at 1.3.1

thornycrackers committed:
maintainers: add thornycrackers

thornycrackers commented:
Updated.

NickCao commented:
@ofborg build python3Packages.python-jwt
